### DISCUSSION TITLE
api key statistics

### DISCUSSION THREAD

**Discussion by ArtemKydr** [apache/skywalking#11597] [Q&A]
Hello everyone There is an application on node.js, how do I build and display a dashboard based on endpoint usage statistics with separate api keys? My app on node.js 16 + postgres
For example, requests:

```
GET /some-endpoint (headers: {x-api-key: "apiKey_1})
GET /some-endpoint (headers: {x-api-key: "apiKey_1})
GET /some-endpoint (headers: {x-api-key: "apiKey_1}) 
GET /some-endpoint (headers: {x-api-key: "apiKey_2}) 
```

Desired statistics:

/some-endpoint (apiKey_1) 3
/some-endpoint (apiKey_2) 1

**Comment by wu-sheng** [comment]
Maybe you need a custom plugin for your HTTP server? @kezhenxu94 Any suggestions?

**Reply by kezhenxu94** [reply]
Agree, this is a customized requirement/feature, a custom plugin is required

**Reply by ArtemKydr** [reply]
Can you help me with it?
Im integrate skywalking in my app like this:
```
if(config['skywalking-agent']) {
    let agent = require('skywalking-backend-js');
    agent.default.start(config['skywalking-agent']);
    console.log('skywalking-agent enabled');
}
```

And all requests catching in middleware like this 
```
module.exports = (req, res, next) => {
    let key = req.query.key || req.headers['x-cartaxi-key'];
    let clientIp = req.headers['x-forwarded-for'] || req.connection.remoteAddress;

    if(!key)
        return res.sendStatus(401);

    getPartnerByKey(key, (err, partner) => {
        if(err) {
            console.error(err);
            return res.sendStatus(401);
        }

        if(!partner){
            return res.sendStatus(401);
        }
}
```

so, I guess that somehow I can add code here to do what I need.
